pkg/auth/registry_remote.go

```diff
@@ -0,0 +1,111 @@
+package auth
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+
+	"crypto/tls"
+
+	"github.com/notaryproject/notation/internal/crypto"
+	"github.com/notaryproject/notation/internal/trace"
+	"github.com/notaryproject/notation/internal/version"
+	"github.com/notaryproject/notation/pkg/config"
+	"github.com/urfave/cli/v2"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+const (
+	PlainHttpName = "plain-http"
+)
+
+type Remote struct {
```

Comment:
Try to enhance `notation-go/registry` instead of having a duplicated implementation in `notation`.